source/Irrlicht/CGUIEditBox.cpp

```diff
@@ -1466,6 +1470,93 @@ void CGUIEditBox::inputChar(wchar_t c)
 	sendGuiEvent(EGET_EDITBOX_CHANGED);
 }
 
+void CGUIEditBox::inputString(core::stringw *str)

```

Comment:
I chose to use pointers since the `StringInput` uses a `core::stringw` pointer.

Reply:
"Bare" instances cannot be put into unions hence it's needed there. Otherwise for functions it's good practice to pass read-only instance arguments as `const T &`.